Leaderboard update

verndrade opened the pull request:
Adds opt-out and makes leaderboard visible again

TODO: Fix UI (make position not fixed), Filter hidden syllabi?

verndrade committed:
opt out leaderboard

winstliu commented:
`self.userprofile_set.annotate(submissions=Count('user__submission', filter=Q(user__submission__hidden=False))).filter(submissions__gt=0).order_by('-submissions')[:5]`

winstliu commented on templates/settings.html:
```diff
@@ -11,7 +25,7 @@ <h4>Delete account</h4>
     <label for="suggestion">Username</label>
     <input class="form-control" id="suggestion" type="text" name="username" required>
-    <small class="form-text{# text-muted #}">Enter your username ({{ user.username }}) to confirm account deletion</small>
+    <small class="form-text{# text-muted #}">Enter your email ({{ user.username }}) to confirm account deletion</small>
```
```suggestion
    <small class="form-text{# text-muted #}">Enter your email ({{ user.email }}) to confirm account deletion</small>
```

winstliu commented on templates/settings.html:
```diff
@@ -11,7 +25,7 @@ <h4>Delete account</h4>
   <div class="form-group">
     <label for="suggestion">Username</label>
```
```suggestion
    <label for="suggestion">Email</label>
```

winstliu commented on templates/settings.html:
```diff
@@ -1,8 +1,22 @@
+<form method="post">
+  {% csrf_token %}
+  {% if user.profile.hide_name %}
+    <p>You are currently Anonymous</p>
```
```suggestion
    <p>Your username is hidden from other users</p>
```

winstliu commented on templates/settings.html:
```diff
@@ -1,8 +1,22 @@
+    <p>You are currently Anonymous</p>
+    <input class="btn btn-success" type="submit" name="opt" value="Show Username">
+  {% else %}
+    <p>You are currently displaying your username: {{user.username | upToCharacter:"@"}}</p>
```
```suggestion
    <p>Your username is visible to other users</p>
```

winstliu commented on templates/settings.html:
```diff
@@ -1,8 +1,22 @@
+  {% csrf_token %}
+  {% if user.profile.hide_name %}
+    <p>You are currently Anonymous</p>
+    <input class="btn btn-success" type="submit" name="opt" value="Show Username">
```
Lowercase username here, same as below

verndrade committed:
add filter for non-approved syllabi and ui changes

winstliu committed:
Merge branch 'development' into verndrade/optOut

mengsarah committed:
better leaderboard semantics and appearance

mengsarah committed:
tweaking aesthetics of leaderboard

mengsarah committed:
making leaderboard not fixed

leaderboard is at the right unless the device is small, in which case the leaderboard is at the bottom

winstliu reviewed (approved):
This is excellent, love the new styling. Only thing I would change is adding a bit more margin when the leaderboard moves below the courses. It looks a little cramped:
![minimal-margin](https://user-images.githubusercontent.com/2766036/90350288-9e65fc00-e00a-11ea-89d5-b73ed1341811.png)

mengsarah commented:
> This is excellent, love the new styling. Only thing I would change is adding a bit more margin when the leaderboard moves below the courses.

I was going to say that I just changed that but oops, for about five minutes I forgot that pull requests and branches are two different things. It's tiny enough I feel like I shouldn't open an entirely new PR though haha.

I went into `index.html` and added the `mb-3` class to the div that holds `<h4>There {{num|pluralize:"is,are"}} {{num}} syllab{{num|pluralize:"us,i"}} for the {{school}}!</h4>`. Adds a bit of extra space to the bottom on large enough screens, too, but I think it still looks okay.
